functions/all_words.py

```
@@ -571,119 +581,166 @@ def smush_all_plan(results, outer_uses, run_budget=1500, time_limit=1.5):
+                return True
+            if state in failed:
+                return False
+            budget[0] -= 1
+            if budget[0] < 0 or time.monotonic() > deadline:
+                budget[0] = -1
+                return False
+            pivot = min((i for i in req if state[i]),
+                        key=lambda i: len(by_letter[i]))
+            options = [s for s in by_letter[pivot]
+                       if used[s] < len(groups[s])
+                       and all(c <= n for c, n in zip(s, state))]
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)